producer_test.go

```diff
@@ -507,25 +509,92 @@ func TestProducerMultipleRetries(t *testing.T) {
 	closeProducer(t, producer)
 }
 
-func ExampleProducer() {
+// This example shows how to use the producer while simultaneously
+// reading the Errors channel to know about any failures.
+func ExampleProducer_select() {
 	producer, err := NewProducer([]string{"localhost:9092"}, nil)
 	if err != nil {
 		panic(err)
 	}
+
 	defer func() {
 		if err := producer.Close(); err != nil {
-			panic(err)
+			log.Fatalln(err)
 		}
 	}()
 
+	// Trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	var enqueued, errors int
 	for {
 		select {
 		case producer.Input() <- &ProducerMessage{Topic: "my_topic", Key: nil, Value: StringEncoder("testing 123")}:
-			fmt.Println("> message queued")
+			enqueued++
 		case err := <-producer.Errors():
-			panic(err.Err)
+			log.Println("Failed to produce message", err)
+			errors++
+		case <-signals:
+			break
```

Comment:
as  the failing test (go vet) caught: you need to break the outer loop, this will just break the select statement by default